src/cmap/commands.ts

```diff
@@ -490,15 +501,66 @@ export class OpMsgRequest {
     return buffers;
   }
 
-  makeDocumentSegment(buffers: Uint8Array[], document: Document): number {
+  /**
+   * Add the sections to the OP_MSG request's buffers and returns the length.
+   */
+  makeSections(buffers: Uint8Array[], document: Document): number {
+    const sequencesBuffer = this.extractDocumentSequences(document);
```

Comment:
The ordering of function calls matter here - extractDocumentSequences must be called before serializeBSON on the document.  I don't think this is covered by a test - could we either ensure that the ordering of calls here is covered or restructuring so ordering isn't important?

Reply:
Yes that's correct ordering is key here so the document sequences get removed from the command before we serialize the first section. I've added unit tests now to ensure that happens.